components/script/dom/xmlhttprequest.rs

```diff
@@ -1038,6 +1044,34 @@ impl XMLHttpRequest {
         encoding.decode(&self.response.borrow(), DecoderTrap::Replace).unwrap().to_owned()
     }
 
+    // https://xhr.spec.whatwg.org/#blob-response
+    fn blob_response(&self) -> Root<Blob> {
+        // Step 1
+        match self.response_blob.get() {
+            Some(response) => response,
+            None => {
+                // Step 2
+                let mime: String = match self.response_headers.borrow().get() {
+                    Some(&ContentType(ref response_mime)) => response_mime.to_string(),
+                    _ => "".to_owned(),
+                };
+
+                // Step 3
+                let bytes = self.response.borrow();
+                let body = if bytes.is_empty() {
```

Comment:
This step here doesn't seem necessary. I think `let blob = Blob::new(self.global().r(), self.response.borrow().to_vec(), &mime);` would be sufficient.